Sphinx job failing in CI due to duplication of `sympy.sets.sets`

All Sphinx jobs are failing with the error message `/home/runner/work/sympy/sympy/doc/src/modules/sets.rst:29: WARNING: duplicate object description of sympy.sets.sets, other instance in modules/sets, use :no-index: for one of them`

There may also be something breaking with the furo them and the new sphinx release:

```
Extension error (furo):
Handler <function _html_page_context at 0x7fa40d613160> for event 'html-page-context' threw an exception (exception: This documentation is not using `furo.css` as the stylesheet. If you have set `html_style` in your conf.py file, remove it.)
make[1]: *** [Makefile:69: _html] Error 2
The docs build failed with the following errors:
/home/moorepants/src/sympy/doc/src/modules/sets.rst:29: WARNING: duplicate object description of sympy.sets.sets, other instance in modules/sets, use :no-index: for one of them

Extension error (furo):
Handler <function _html_page_context at 0x7fa40d613160> for event 'html-page-context' threw an exception (exception: This documentation is not using `furo.css` as the stylesheet. If you have set `html_style` in your conf.py file, remove it.)
```

--- a/doc/requirements.txt
+++ b/doc/requirements.txt
@@ -5,7 +5,7 @@ matplotlib-inline
 mpmath
 myst-parser
 scipy
-Sphinx
+Sphinx<7.2.0
 sphinx-autobuild
 sphinx-copybutton
 sphinx-math-dollar

